lighthouse-core/scripts/lantern/collect/collect.js

```diff
@@ -0,0 +1,316 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @typedef {{trace: string}} Result */
+/** @typedef {{url: string, wpt: Result[], unthrottled: Result[]}} UrlResults */
+
+const archiver = require('archiver');
+const path = require('path');
+const fs = require('fs');
+const readline = require('readline');
+const https = require('https');
+const {execFile} = require('child_process');
+
+const LH_ROOT = path.join(__dirname, '..', '..', '..', '..');
+const SAMPLES = 9;
+const URLS = require('./urls.json');
+
+if (!process.env.WPT_KEY) throw new Error('missing WPT_KEY');
+const WPT_KEY = process.env.WPT_KEY;
+const DEBUG = process.env.DEBUG;
+
+const outputFolder = path.join(LH_ROOT, 'dist', 'lantern-traces');
+const summaryPath = path.join(outputFolder, 'summary.json');
+
+class ProgressLogger {
+  constructor() {
+    this._currentProgressMessage = '';
+    this._loadingChars = '⣾⣽⣻⢿⡿⣟⣯⣷ ⠁⠂⠄⡀⢀⠠⠐⠈';
+    this._nextLoadingIndex = 0;
+    this._progressBarHandle = setInterval(() => this.progress(this._currentProgressMessage), 100);
+  }
+
+  /**
+   * @param  {...any} args
+   */
+  log(...args) {
+    readline.clearLine(process.stdout, 0);
+    readline.cursorTo(process.stdout, 0);
+    // eslint-disable-next-line no-console
+    console.log(...args);
+    this.progress(this._currentProgressMessage);
+  }
+
+  /**
+   * @param {string} message
+   */
+  progress(message) {
+    this._currentProgressMessage = message;
+    readline.clearLine(process.stdout, 0);
+    readline.cursorTo(process.stdout, 0);
+    if (message) process.stdout.write(`${this._nextLoadingChar()} ${message}`);
+  }
+
+  close() {
+    clearInterval(this._progressBarHandle);
+    this.progress('');
+  }
+
+  _nextLoadingChar() {
+    const char = this._loadingChars[this._nextLoadingIndex++];
+    if (this._nextLoadingIndex >= this._loadingChars.length) {
+      this._nextLoadingIndex = 0;
+    }
+    return char;
+  }
+}
+
+/**
+ *
+ * @param {string} archiveDir
+ * @param {string} outputPath
+ */
+function archive(archiveDir, outputPath) {
+  return new Promise((resolve, reject) => {
+    const archive = archiver('zip', {
+      zlib: {level: 9},
+    });
+
+    const writeStream = fs.createWriteStream(outputPath);
+    writeStream.on('finish', resolve);
+    writeStream.on('error', reject);
+
+    archive.pipe(writeStream);
+    archive.directory(archiveDir, false);
+    archive.finalize();
+  });
+}
+
+const log = new ProgressLogger();
```

Comment:
nit: should be constructed in main()? (`summary` probably could too)

Reply:
log needs to be globally accessible

Reply:
or do you just mean `let log;` and instantiate in main?

Reply:
> or do you just mean let log; and instantiate in main?

:+1: